### PR TITLE
Fix flaky comment timestamp spec

### PR DESCRIPTION
**Pull request by jacobherrington**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [x] Optimization
- [ ] Documentation Update

## Description

This spec is flakey. There are a couple of ways to solve this quickly:
1) ~Just test for the CSS and don't worry about the timestamp accuracy (we could just delete the test as other tests check for the presence of the timestamp's parent element.~
2) We can test the timestamp against a regex and not worry about the accuracy of the timestamp, just to be sure *some* timestamp shows up.
3) ~Verify that the timestamp is rendered and matches at least the same day as we expect it to be -- which should give us a little confidence, but not be as flaky.~

After getting some feedback from Molly, I went the regex direction! :capital_abcd: 

## Added tests?

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] I've updated the README or added inline documentation
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [x] This change does not need to be communicated, and this is why not: test fix!

### CODE REVIEW

**Comment by mstruve** [comment]
In my experience the datetime is usually completely off and even doing beginning of day will be flaky bc you may have a difference of day between the locale server time and the rspec random time zone we choose. More info here when I tried to fix it: https://github.com/forem/forem/pull/11025 and then when I gave up https://github.com/forem/forem/pull/11087

**Comment by jacobherrington** [comment]
@mstruve That makes sense! How do you feel about using regex to ensure that we are at least rendering *a timestamp* in the correct format? I did that first but swapped to this approach.

Otherwise, I think we should just delete the test case because other tests will fail if the element isn't rendered at all.

What do you think?

**Comment by mstruve** [comment]
> How do you feel about using regex to ensure that we are at least rendering a timestamp in the correct format?

As long as I am not the one writing the regex I'm totally game!!! 😆

File: spec/system/user/view_user_index_spec.rb [hunk 1]
```diff
@@ -63,9 +63,9 @@ def shows_comments
 
       def shows_comment_timestamp
         within("#substories .profile-comment-card .profile-comment-row:first-of-type") do
-          ts = comment.decorate.published_timestamp
-          timestamp_selector = ".comment-date time[datetime='#{ts}']"
-          expect(page).to have_selector(timestamp_selector)
+          iso8601_date_time = /^((\d{4})-(\d{2})-(\d{2})T(\d{2}):(\d{2}):(\d{2})Z)$/
```
Comment (jacobherrington) [inline spec/system/user/view_user_index_spec.rb:66]: Not being too strict here, but just verifying that we're including the right number of digits in the right order.

image

**Review by rhymes — APPROVED** [review]
Let's try this!

**Review by maestromac — APPROVED** [review]

**Pull request merged**